pkg/clean/clean_lb_infra.go

```diff
@@ -59,14 +59,12 @@ func (s *LBInfraCleaner) CleanupInfraResources(ctx context.Context) error {
 		s.log.Error(err, "Failed to clean up DLB virtual servers")
 		return err
 	}
-	// SharedResource has dependencies on Group, so we can't delete sharedResources and groups in parallel.
-	if err := s.cleanupInfraSharedResources(ctx); err != nil {
-		s.log.Error(err, "Failed to clean up infra SharedResources")
+	if err := s.cleanupInfraShares(ctx); err != nil {
```

Comment:
Why do we move cleanupInfraShares from parallelCleaners here?

Reply:
Do we need to remove
```
case *model.SharedResource:
		return *v.Path, nil
```
from keyFunc?

Reply:
As the comments said, 'we can't delete sharedResources and groups in parallel', I'm not sure if we could delete the shares and grups in parallel, so I try to cleanupInfraShares first.
For the case *model.SharedResource, it seems that's ok to have it. Not sure if later we'll use it.